scripts/travis/upload_coverage.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Download codecov bash script, verify checksum from github, run script
+
+# fail if shasum detects a problem
+set -e
```

Comment:
you're using pipes here, so I'd suggest `set -o pipefail` to ensure we don't mask out errors.